libcxx/include/__memory/aligned_alloc.h

```diff
@@ -30,7 +30,7 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 inline _LIBCPP_HIDE_FROM_ABI void* __libcpp_aligned_alloc(std::size_t __alignment, std::size_t __size) {
 #  if defined(_LIBCPP_MSVCRT_LIKE)
   return ::_aligned_malloc(__size, __alignment);
-#  elif _LIBCPP_STD_VER >= 17 && !defined(_LIBCPP_HAS_NO_C11_ALIGNED_ALLOC)
+#  elif defined(_LIBCPP_HAS_C11_ALIGNED_ALLOC) || (_LIBCPP_STD_VER >= 17 && !defined(_LIBCPP_HAS_NO_C11_ALIGNED_ALLOC))
```

Comment:
Is there a reason for introducing a new macro named `_LIBCPP_HAS_C11_ALIGNED_ALLOC`? It should suffice not to define `_LIBCPP_HAS_NO_C11_ALIGNED_ALLOC` on picolibc, no?

Reply:
The latter is already not defined, so this code will select C11 `aligned_alloc` with picolibc as long as `_LIBCPP_STD_VER >= 17`.

This change is intended to make it select `aligned_alloc` even when that's _not_ true: we can't fall back to any other API because it doesn't reliably have them.

Reply:
> This change is intended to make it select aligned_alloc even when that's not true

You mean use `aligned_alloc` in C++ < 17? I'd rather not do that. We want to avoid relying on non-Standard behavior (or introducing any complexity for that purpose), and we can't be guaranteed that the C library provides `aligned_alloc` before C++17.

I think it's reasonable to require that picolibc users compile with C++ >= 17 when they use an allocator that doesn't provide `posix_memalign`.